uc-crux-llvm/src/UCCrux/LLVM/Postcondition/Type.hs

```diff
@@ -0,0 +1,275 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeOperators #-}
+
+{-
+Module       : UCCrux.LLVM.Postcondition.Type
+Description  : Postconditions for LLVM functions
+Copyright    : (c) Galois, Inc 2021
+License      : BSD3
+Maintainer   : Langston Barrett <langston@galois.com>
+Stability    : provisional
+
+A postcondition for a function summarizes its return value and its effect of a
+function on LLVM memory. Of course, the most complete summary of a function is
+semantically equivalent to its implementation. The postconditions in this module
+are significantly less expressive than that. The aims are to support
+postcondition inference, and to support partial (user- or programmer-supplied)
+specifications of external functions.
+
+'ClobberGlobal' and 'ClobberArg' can explicitly set a type of data to generate,
+which may or may not actually match the declared type of the data in question.
+This is useful for clobbering e.g. a @char*@ or @void*@ with structured data.
+However, it yields some API and implementation complexity due to possible type
+mismatches (see 'OpaquePointers').
```

Comment:
I've added another paragraph that goes into more detail on this, let me know if it helps!

Reply:
> This paragraph is slightly suggestive, but can you add a more explicit blurb between these paragraphs about what is captured in a postcondition? It seems like it might be "this location or argument is clobbered".

I figured this would be captured by the docs on the fields of `UPostcond` and `Postcond` - Should I perhaps summarize those here and provide a link to those? Or is there a different level of detail/granularity that would be helpful here?

> These are then not postconditions on the return value or even really the post-state, but are more like over-approximate summaries of program behavior?

They include both postconditions on function return values and also information about clobbered globals and arguments. And they can be overapproximate or underapproximate, depending on how they're written! UC-Crux will deduce over-approximate conditions on the return values, and currently doesn't deduce other parts of the `Postcond` structure (but we need it to down the line - in particular, we need it to deduce conditions on clobbered arguments).